core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -1040,4 +1047,195 @@ public void testAllManifestsTableSnapshotNot() {
         expectedManifestListPaths(table.snapshots(), 1L, 3L, 4L),
         actualManifestListPaths(manifestsTableScan));
   }
+
+  @Test
+  public void testPositionDeletesWithFilter() {
+    Assume.assumeTrue("Position deletes supported only for v2 tables", formatVersion == 2);
+    preparePartitionedTable();
+
+    PositionDeletesTable positionDeletesTable = new PositionDeletesTable(table);
+
+    Expression expression =
+        Expressions.and(
+            Expressions.equal("partition.data_bucket", 1), Expressions.greaterThan("pos", 0));
+    BatchScan scan = positionDeletesTable.newBatchScan().filter(expression);
+    assertThat(scan).isExactlyInstanceOf(PositionDeletesTable.PositionDeletesBatchScan.class);
+
+    List<ScanTask> tasks = Lists.newArrayList(scan.planFiles());
+
+    Assert.assertEquals(
+        "Expected to scan one delete manifest",
+        1,
+        ((PositionDeletesTable.PositionDeletesBatchScan) scan)
+            .scanMetrics()
+            .scannedDeleteManifests()
+            .value());
+    Assert.assertEquals(
+        "Expected to skip three delete manifests",
+        3,
+        ((PositionDeletesTable.PositionDeletesBatchScan) scan)
```

Comment:
It may be easier to read if you did the cast prior to this call.

Reply:
Done